src/quo2/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,296 @@
+                                                             (get-colors "neutral-5")))]
+        box-style)
+      :align-self "flex-end"
+      :border-color
+      (get-colors "neutral-10"))]]])
+
+(defn render-text
+  [title override-theme &
+   {:keys [typography weight size style]
+    :or   {typography :main-semibold
+           weight     :semi-bold
+           size       :paragraph-1
+           style      (style/title override-theme)}}]
+  [text/text
+   {:typography          typography
+    :accessibility-label :title-name-text
+    :ellipsize-mode      :tail
+    :style               style
+    :override-theme      override-theme
+    :number-of-lines     1
+    :weight              weight
+    :size                size}
+   title])
+
+(defn network-type-text
+  [networkType networkState]
+  (cond
+    (or (= networkState "sending") (= networkState "pending"))      "Pending on"
+    (or (= networkState "confirmed") (= networkState "finalising")) "Confirmed on"
+    (= networkState "finalised")                                    "Finalised on"
+    (= networkState "error")                                        "Failed on"))
+
+(defn steps-text
+  [networkType networkState]
+  (cond
+    (and (= networkType "mainnet")
+         (not= networkState "finalised")
+         (not= networkState "error"))   (str (if (< (@app-state :counter) 4)
+                                               (@app-state :counter)
+                                               "4")
+                                             "/4")
+    (= networkState "finalised")        "Epoch 181,329"
+    (and (= networkType "mainnet")
+         (= networkState "error"))      "0/4"
+    (and (= networkType "optimism/arbitrum")
+         (= networkState "finalising")) "1/1"
+    (= networkType "optimism/arbitrum") "0/1"))
+
+(defn get-status-icon
+  [networkType networkState]
+  (cond
+    (or (= networkState "pending") (= networkState "sending"))      ["pending-state"
+                                                                     (get-colors "neutral-50")]
+    (or (= networkState "confirmed") (= networkState "finalising")) ["positive-state"
+                                                                     (get-colors "success-50")]
+    (= networkState "finalised")                                    ["diamond" (get-colors "success-50")]
+    (= networkState "error")                                        ["negative-state"
+                                                                     (get-colors "danger-50")]))
+
+(defn transaction-progress
+  [{:keys [title
+           on-press
+           accessibility-label
+           networkType
+           networkState
+           container-style
+           override-theme]}]
+  (let [count (reagent/atom 0)]
+    (rn/use-effect
+     (fn []
+       (start-interval networkState)
+       (clear-counter)
+       (fn []
+         (stop-interval)))
+     [networkState])
```

Comment:
I believe we don't need this interval inside the component. `status-go` will send a signal update on the transaction, and we can update the state from outside the component as it's the best practice to make the component more modular/independent.

Reply:
thanks